vxsandbox/resources/tests/test_metrics.py

```diff
@@ -0,0 +1,123 @@
+        self.assertEqual(ev, MetricEvent('mystore', 'foo', 1.5, self.SUM))
+
+    def test_bad_store(self):
+        self.assertRaises(MetricEventError, MetricEvent.from_command, {
+                'store': 'foo bar', 'metric': 'foo', 'value': 1.5,
+                'agg': 'sum'})
+
+    def test_bad_type_store(self):
+        self.assertRaises(MetricEventError, MetricEvent.from_command, {
+                'store': {}, 'metric': 'foo', 'value': 1.5,
+                'agg': 'sum'})
+
+    def test_bad_metric(self):
+        self.assertRaises(MetricEventError, MetricEvent.from_command, {
+                'store': 'mystore', 'metric': 'foo bar', 'value': 1.5,
+                'agg': 'sum'})
+
+    def test_bad_type_metric(self):
+        self.assertRaises(MetricEventError, MetricEvent.from_command, {
+                'store': 'mystore', 'metric': {}, 'value': 1.5,
+                'agg': 'sum'})
+
+    def test_missing_metric(self):
+        self.assertRaises(MetricEventError, MetricEvent.from_command, {
+                'store': 'mystore', 'value': 1.5, 'agg': 'sum'})
+
+    def test_bad_value(self):
+        self.assertRaises(MetricEventError, MetricEvent.from_command, {
+                'store': 'mystore', 'metric': 'foo', 'value': 'abc',
+                'agg': 'sum'})
+
+    def test_bad_type_value(self):
+        self.assertRaises(MetricEventError, MetricEvent.from_command, {
+                'store': 'mystore', 'metric': 'foo', 'value': {},
+                'agg': 'sum'})
+
+    def test_missing_value(self):
+        self.assertRaises(MetricEventError, MetricEvent.from_command, {
+                'store': 'mystore', 'metric': 'foo', 'agg': 'sum'})
+
+    def test_bad_agg(self):
+        self.assertRaises(MetricEventError, MetricEvent.from_command, {
+                'store': 'mystore', 'metric': 'foo', 'value': 1.5,
+                'agg': 'foo'})
+
+    def test_bad_type_agg(self):
+        self.assertRaises(MetricEventError, MetricEvent.from_command, {
+                'store': 'mystore', 'metric': 'foo', 'value': 1.5,
+                'agg': {}})
+
+    def test_missing_agg(self):
+        self.assertRaises(MetricEventError, MetricEvent.from_command, {
+                'store': 'mystore', 'metric': 'foo', 'value': 1.5})
+
+
+class TestMetricsResource(VumiTestCase):
+
+    SUM = MetricEvent.AGGREGATORS['sum']
+
+    def setUp(self):
+        self.dummy_api = object()
+
+    def check_reply(self, reply, cmd, success):
+        self.assertEqual(reply['reply'], True)
+        self.assertEqual(reply['cmd_id'], cmd['cmd_id'])
+        self.assertEqual(reply['success'], success)
+
+    def check_publish(self, store, metric, value, agg):
+        self.app_worker.publish_account_metric.assert_called_once_with(
+            self.conversation.user_account.key, store, metric, value, agg)
+
+    def check_not_published(self):
+        self.assertFalse(self.app_worker.publish_account_metric.called)
```

Comment:
If you replace `check_publish` and `check_not_publish` so that they are just:
```
"""
pass  # TODO: check that a metric is published
"""
```
then you should be able to put back the bodies of the test functions that you deleted.